File: JavaScript/leetcode75/src/main/Problem04_CanPlaceFlowers.js
/**
 * @param {number[]} flowerbed
 * @param {number} n
 * @return {boolean}
 */
var canPlaceFlowers = function (flowerbed, n) {
  for (let i = 0; i < flowerbed.length; i++) {
    if (!flowerbed[i - 1] && !flowerbed[i] && !flowerbed[i + 1]) {
      n--;
      flowerbed[i] = 1;
    }
    if (n === 0) {
      return true;
    }
  }
  return n <= 0;
};

module.exports = canPlaceFlowers;

File: Java/leetcode75/src/main/java/org/phoenix/leetcode/challenges/Problem04_CanPlaceFlowers.java
package org.phoenix.leetcode.challenges;

public class Problem04_CanPlaceFlowers {
    public boolean canPlaceFlowers(int[] flowerbed, int n){
        int count = 0;

        for(int i = 0; i < flowerbed.length; i++){
            if(flowerbed[i] == 0){
                int prev = (i == 0 || flowerbed[i - 1] == 0) ? 0 : 1;
                int next = (i == flowerbed.length - 1 || flowerbed[i + 1] == 0) ? 0 : 1;

                if(prev == 0 && next == 0){
                    flowerbed[i] = 1;
                    count++;
                }
            }
        }
        return count >= n;
    }
}
